src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1903,6 +1921,47 @@ public void ShowSaveDialogAndSaveResult(object parameter)
             }
         }
 
+        private bool ShowWarningDialogOnSaveWithUnresolvedIssues()
+        {
+            if (Model.LinterManager.RuleEvaluationResults.Count <= 0)
+            {
+                return true;
+            }
+
+            string imageUri = "/DynamoCoreWpf;component/UI/Images/task_dialog_future_file.png";
+            var args = new TaskDialogEventArgs(
+                new Uri(imageUri, UriKind.Relative), 
+                WpfResources.GraphIssuesOnSave_Title,
+                WpfResources.GraphIssuesOnSave_Summary, 
+                WpfResources.GraphIssuesOnSave_Description);
+
+            args.AddRightAlignedButton((int)DynamoModel.ButtonId.Proceed, WpfResources.GraphIssuesOnSave_ProceedBtn);
+            args.AddRightAlignedButton((int)DynamoModel.ButtonId.Cancel, WpfResources.GraphIssuesOnSave_CancelBtn);
+
+            var dialog = new GenericTaskDialog(args);
+
+            if (DynamoModel.IsTestMode)
```

Comment:
Overall I think this looks good. 
For the test I'm ok with it as it is but I'm not sure how much value we get since we're not going through the same route when user is at the helm and we're adding some code to make that happen.
How about we just test if the LinterView has opened when there are issues and cancel was pressed ?
I'm thinking to do this : we trigger an event for the cancel action for which we have a handler that will check issues and will open linter view if it has to ( this event will be triggered from both user mode and test mode ).
And this would be the testable part, we will just raise this event from the test, no pressing on Cancel button since we don't have access/can't control it.